client/signup/steps/site-type/index.jsx

```diff
@@ -129,10 +129,11 @@ export default connect(
 			let nextFlowName = ownProps.flowName;
 			if ( siteTypeValue === getSiteTypePropertyValue( 'id', 'store', 'slug' ) ) {
 				nextFlowName = 'ecommerce';
-			} else if ( 'ecommerce' === ownProps.flowName && ownProps.previousFlowName ) {
-				nextFlowName = ownProps.previousFlowName;
 			}
 
+			if ( siteTypeValue === getSiteTypePropertyValue( 'id', 'business', 'slug' ) ) {
```

Comment:
Nitpick: this should be `else if`, otherwise this conditional statement will still be evaluated even if the previous one is true.